Jellyfin.Server.Implementations/Users/UserManager.cs

```diff
@@ -0,0 +1,787 @@
+
+            var dbContext = _dbProvider.CreateContext();
+
+            if (dbContext.Users.Find(user.Id) == null)
+            {
+                throw new ArgumentException(string.Format(
+                    CultureInfo.InvariantCulture,
+                    "The user cannot be deleted because there is no user with the Name {0} and Id {1}.",
+                    user.Username,
+                    user.Id));
+            }
+
+            if (dbContext.Users.Count() == 1)
+            {
+                throw new InvalidOperationException(string.Format(
+                    CultureInfo.InvariantCulture,
+                    "The user '{0}' cannot be deleted because there must be at least one user in the system.",
+                    user.Username));
+            }
+
+            if (user.HasPermission(PermissionKind.IsAdministrator)
+                && Users.Count(i => i.HasPermission(PermissionKind.IsAdministrator)) == 1)
+            {
+                throw new ArgumentException(
+                    string.Format(
+                        CultureInfo.InvariantCulture,
+                        "The user '{0}' cannot be deleted because there must be at least one admin user in the system.",
+                        user.Username),
+                    nameof(user));
+            }
+
+            dbContext.Users.Remove(user);
+            dbContext.SaveChanges();
+            OnUserDeleted?.Invoke(this, new GenericEventArgs<User>(user));
+        }
+
+        /// <inheritdoc/>
+        public Task ResetPassword(User user)
+        {
+            return ChangePassword(user, string.Empty);
+        }
+
+        /// <inheritdoc/>
+        public void ResetEasyPassword(User user)
+        {
+            ChangeEasyPassword(user, string.Empty, null);
+        }
+
+        /// <inheritdoc/>
+        public async Task ChangePassword(User user, string newPassword)
+        {
+            if (user == null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+
+            await GetAuthenticationProvider(user).ChangePassword(user, newPassword).ConfigureAwait(false);
+            await UpdateUserAsync(user).ConfigureAwait(false);
+
+            OnUserPasswordChanged?.Invoke(this, new GenericEventArgs<User>(user));
+        }
+
+        /// <inheritdoc/>
+        public void ChangeEasyPassword(User user, string newPassword, string newPasswordSha1)
+        {
+            GetAuthenticationProvider(user).ChangeEasyPassword(user, newPassword, newPasswordSha1);
+            UpdateUser(user);
+
+            OnUserPasswordChanged?.Invoke(this, new GenericEventArgs<User>(user));
+        }
+
+        /// <inheritdoc/>
+        public UserDto GetUserDto(User user, string remoteEndPoint = null)
+        {
+            return new UserDto
+            {
+                Name = user.Username,
+                Id = user.Id,
+                ServerId = _appHost.SystemId,
+                HasPassword = user.Password != null,
```

Comment:
```suggestion
            var authenticationProvider = GetAuthenticationProvider(user);
            var hasPassword = authenticationProvider.HasPassword(user);
            var hasEasyPassword = !string.IsNullOrEmpty(authenticationProvider.GetEasyPasswordHash(user));
            return new UserDto
            {
                Name = user.Username,
                Id = user.Id,
                ServerId = _appHost.SystemId,
                HasPassword = hasPassword,
                HasConfiguredPassword = hasPassword,
                HasConfiguredEasyPassword = hasEasyPassword,
```

`HasPassword` is false if user is using an authentication provider.
WebUI requires both `HasPassword` and `HasConfiguredPassword`
`HasEasyPassword` wasn't being set.